pkg/agent/util/net_windows.go

```diff
@@ -521,29 +521,44 @@ func PrepareHNSNetwork(subnetCIDR *net.IPNet, nodeIPNet *net.IPNet, uplinkAdapte
 
 // adapterIPExists finds the network adapter configured with the provided IP, MAC and its name has the given prefix.
 // If "namePrefix" is empty, it returns the first network adapter with the provided IP and MAC.
+// If DHCP is enabled on the original adapter, it will attempt to acquire an IP multiple times.
 // It returns true if the IP is found on the adapter, otherwise it returns false.
-func adapterIPExists(ip net.IP, mac net.HardwareAddr, namePrefix string) (*net.Interface, bool, error) {
+func adapterIPExists(ip net.IP, mac net.HardwareAddr, namePrefix string, dhcp bool) (*net.Interface, bool, error) {
 	adapters, err := netInterfaces()
 	if err != nil {
 		return nil, false, err
 	}
+	// To avoid the issue of insufficient time for the Windows adapter to obtain an dhcp IP after the creation of the HNS network,
+	// wait for up to three seconds and retry IP acquisition in case of failures.
 	ipExists := false
 	for idx, adapter := range adapters {
 		if bytes.Equal(adapter.HardwareAddr, mac) {
 			if namePrefix == "" || strings.Contains(adapter.Name, namePrefix) {
-				addrList, err := netInterfaceAddrs(&adapters[idx])
-				if err != nil {
-					return nil, false, err
-				}
-				for _, addr := range addrList {
-					if ipNet, ok := addr.(*net.IPNet); ok {
-						if ipNet.IP.Equal(ip) {
-							ipExists = true
-							break
+				for retry := 0; retry < 4; retry++ {
+					addrList, err := netInterfaceAddrs(&adapters[idx])
+					if err != nil {
+						return nil, false, err
+					}
+					for _, addr := range addrList {
+						if ipNet, ok := addr.(*net.IPNet); ok {
+							if ipNet.IP.Equal(ip) {
+								ipExists = true
+								break
+							}
 						}
 					}
+					if ipExists {
+						return &adapter, true, nil
+					}
+					if dhcp == false {
+						return &adapter, false, nil
+					} else if !ipExists && retry < 4 {
+						time.Sleep(1 * time.Second)
+					} else {
+						klog.Warning("Unable to find an available DHCP IP with MAC %s, IP %s, and name prefix %s after retries", mac.String(), ip.String(), namePrefix)
```

Comment:
You can move this check/warning outside of the retry.

Reply:
Updated.